darshan-util/pydarshan/tests/test_summary.py

```diff
@@ -122,3 +125,229 @@ def test_main_all_logs_repo_files(tmpdir, log_repo_files):
+                "16",
+                "< 1",
+                "/yellow/usr/projects/eap/users/treddy/simple_dxt_mpi_io_darshan/a.out",
+                "3.2.1",
+                "3.21",
+            ),
+        ],
+    )
+    def test_metadata_table(
+        self,
+        filename,
+        expected_jobid,
+        expected_uid,
+        expected_nprocs,
+        expected_runtime,
+        expected_cmd,
+        expected_lib_ver,
+        expected_log_fmt,
+    ):
+        # regression test for `summary.ReportData.get_metadata_table()`
+
+        # generate the report data
+        log_path = os.path.join("tests/input/", filename)
+        R = summary.ReportData(log_path=log_path)
+        # convert the metadata table back to a pandas dataframe
+        actual_metadata_df = pd.read_html(R.metadata_table, index_col=0)[0]
+
+        # check the metadata match the expected outcomes
+        assert actual_metadata_df.loc["Job ID"].values[0] == expected_jobid
+        assert actual_metadata_df.loc["User ID"].values[0] == expected_uid
+        assert actual_metadata_df.loc["# Processes"].values[0] == expected_nprocs
+        assert actual_metadata_df.loc["Runtime (s)"].values[0] == expected_runtime
+        assert actual_metadata_df.loc["Command"].values[0] == expected_cmd
+        assert actual_metadata_df.loc["Log Filename"].values[0] == filename
+        assert (
+            actual_metadata_df.loc["Runtime Library Version"].values[0]
+            == expected_lib_ver
+        )
+        assert actual_metadata_df.loc["Log Format Version"].values[0] == expected_log_fmt
+
+        # get the time stamps in the current time zone
+        expected_start = str(datetime.fromtimestamp(R.report.metadata["job"]["start_time"]))
+        expected_end = str(datetime.fromtimestamp(R.report.metadata["job"]["end_time"]))
+        assert actual_metadata_df.loc["Start Time"].values[0] == expected_start
+        assert actual_metadata_df.loc["End Time"].values[0] == expected_end
+
+
+    @pytest.mark.parametrize(
+        "log_path",
+        [
+            # each of these logs offers a unique
+            # set of modules to verify
+            "tests/input/sample.darshan",
+            "tests/input/noposix.darshan",
+            "tests/input/noposixopens.darshan",
+            "tests/input/sample-goodost.darshan",
+            "tests/input/sample-dxt-simple.darshan",
+        ],
+    )
+    def test_module_table(self, log_path):
+        # regression test for `summary.ReportData.get_module_table()`
+
+        # collect the report data
+        R = summary.ReportData(log_path=log_path)
+        # convert the module table back to a pandas dataframe
+        actual_mod_df = pd.read_html(R.module_table, index_col=0)[0]
+
+        # verify the number of modules in the report is equal to
+        # the number of rows in the module table
+        expected_module_count = len(R.report.modules.keys())
+        assert expected_module_count == actual_mod_df.shape[0]
```

Comment:
any reason to place the expected value first here, and second almost everywhere else?